### PR TITLE
feat(looker): represent looker dashboards as assets

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
As the title. We'll represent the views as assets next.

Also, probably won't use a `multi_asset` decorator since these are "external" assets (naming tbd), so that will also probably change. But that's an easy refactor later.

## How I Tested These Changes
pytest, local

Screenshot 2024-04-18 at 4 35 43 PM

### CODE REVIEW

**Comment by rexledesma** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21294?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#21294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21294?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#21293** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21293?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21284** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21284?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+def looker_assets(*, project_dir: Path) -> Callable[[Callable[..., Any]], AssetsDefinition]:
+    lookml_dashboard_specs = [
+        AssetSpec(
+            key=AssetKey(lookml_dashboard["dashboard"]),
+            deps={AssetKey(dashboard_element["explore"])},
+        )
+        for dashboard_path in project_dir.rglob("*.dashboard.lookml")
+        # Each dashboard file can contain multiple dashboards.
+        for lookml_dashboard in yaml.safe_load(dashboard_path.read_bytes())
```
Comment (benpankow) [inline python_modules/libraries/dagster-looker/dagster_looker/asset_decorator.py:17]: at first glance the triple-comprehension was a little hard to follow, especially with the filter condition, might be easier to parse split into a few intermediate steps (& would probably be more self-documenting/reduce need for inline comments)

**Review by benpankow — APPROVED** [review]
Looks good

**Comment by rexledesma** [comment]
### Merge activity

* **May 2, 9:50 AM PDT**: [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/21294) rebased this pull request as part of a merge.
